drivers/SmartThings/zigbee-switch/src/test/test_aqara_smart_plug.lua

```diff
@@ -0,0 +1,274 @@
+    test.socket.zigbee:__set_channel_ordering("relaxed")
+
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      zigbee_test_utils.build_bind_request(mock_device,
+        zigbee_test_utils.mock_hub_eui,
+        SimpleMetering.ID)
+    })
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      SimpleMetering.attributes.InstantaneousDemand:configure_reporting(mock_device, 1, 3600, 1)
+    })
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      SimpleMetering.attributes.CurrentSummationDelivered:configure_reporting(mock_device, 5, 3600, 1)
+    })
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      zigbee_test_utils.build_bind_request(mock_device,
+        zigbee_test_utils.mock_hub_eui,
+        ElectricalMeasurement.ID)
+    })
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      ElectricalMeasurement.attributes.ActivePower:configure_reporting(mock_device, 1, 3600, 1)
+    })
+    test.socket.zigbee:__expect_send(
+      {
+        mock_device.id,
+        Groups.server.commands.RemoveAllGroups(mock_device)
+      }
+    )
+
+    mock_device:expect_metadata_update({ provisioning_state = "PROVISIONED" })
+  end
+)
+
+test.register_message_test(
+  "Capability command On should be handled",
+  {
+    {
+      channel = "capability",
+      direction = "receive",
+      message = { mock_device.id, { capability = "switch", component = "main", command = "on", args = {} } }
+    },
+    {
+      channel = "zigbee",
+      direction = "send",
+      message = { mock_device.id, OnOff.server.commands.On(mock_device) }
+    }
+  }
+)
+
+test.register_message_test(
+  "Capability command Off should be handled",
+  {
+    {
+      channel = "capability",
+      direction = "receive",
+      message = { mock_device.id, { capability = "switch", component = "main", command = "off", args = {} } }
+    },
+    {
+      channel = "zigbee",
+      direction = "send",
+      message = { mock_device.id, OnOff.server.commands.Off(mock_device) }
+    }
+  }
+)
+
+test.register_message_test(
+  "Handle Power meter",
+  {
+    {
+      channel = "zigbee",
+      direction = "receive",
+      message = { mock_device.id,
+        SimpleMetering.attributes.InstantaneousDemand:build_test_attr_report(mock_device, 64) }
+    },
+    {
+      channel = "capability",
+      direction = "send",
+      message = mock_device:generate_test_message("main", capabilities.powerMeter.power({ value = 6.4, unit = "W" }))
+    }
+  }
+)
+
+test.register_message_test(
+  "Handle Energy meter",
+  {
+    {
+      channel = "zigbee",
+      direction = "receive",
+      message = { mock_device.id,
+        SimpleMetering.attributes.CurrentSummationDelivered:build_test_attr_report(mock_device, 0x15a) }
+    },
+    {
+      channel = "capability",
+      direction = "send",
+      message = mock_device:generate_test_message("main",
+        capabilities.energyMeter.energy({ value = 0.346, unit = "kWh" }))
+    }
```

Comment:
shouldn't there also be a PowerConsumptionReport event after this?

Reply:
I added PowerConsumptionReport test code. :)